Only log.warn once on feed failure

aledsage opened the pull request.

drigodwin commented:
```
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] /home/jenkins/jenkins-slave/workspace/brooklyn-server-pull-requests/test-framework/src/test/java/org/apache/brooklyn/test/framework/TestSensorTest.java:[390,39] method getEvents in class org.apache.brooklyn.test.LogWatcher cannot be applied to given types;
  required: no arguments
  found: com.google.common.base.Predicate<ch.qos.logback.classic.spi.ILoggingEvent>
  reason: actual and formal argument lists differ in length
```

aledsage commented:
Thanks @geomacy @drigodwin - I'd accidentally deleted the `LogWatcher.getEvents(Predicate)` method, so I've reverted that (and squashed into a single commit)!
